### PR TITLE
Readiness

### PR DESCRIPTION
**Pull request by yuval-k**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/872

### CODE REVIEW

File: changelog/v0.17.7/readiness.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: NEW_FEATURE
+    description: Add liveness and readiness probes
+    issueLink: https://github.com/solo-io/gloo/issues/872
```
Comment (kdorosh) [inline changelog/v0.17.7/readiness.yaml]: before we mark this as resolved, can we add similar checks to our other three deployments?

Comment (yuval-k) [reply changelog/v0.17.7/readiness.yaml]: done

Comment (yuval-k) [reply changelog/v0.17.7/readiness.yaml]: by three you meant two, right?

Comment (kdorosh) [reply changelog/v0.17.7/readiness.yaml]: i meant:
- gloo
- gateway
- gateway-proxy
- discovery

But including ingress, we should really handle all 6, thus we should have checks on:
- ingress-deployment
- ingress-proxy-deployment

as well

Comment (yuval-k) [reply changelog/v0.17.7/readiness.yaml]: adding readiness to gloo is a good idea, but should probably be done in separate PR; as ideally, you only want to declare gloo as ready once it has done at least one sync loop and has a snapshot ready for envoy.

**Comment by yuval-k** [comment]
@rickducott  i think this is a legit fail; looking into this

**Comment by yuval-k** [comment]
probes should work now; had to remove them from ingress and cluster ingress until https://github.com/solo-io/gloo/pull/822 is merged

**Pull request merged**